nservicebus/hosting/nservicebus-host/troubleshooting.md

```diff
@@ -34,17 +34,21 @@ Verify that:
 1. Assembly binding redirects in the `NServiceBus.Host.exe.config` are correct and match with any entries generated in the `App.config` file.
 2. All files in the deployment (sub)folder are correct and no old files are present from previous deployments.
 
-## Service fails to start due to reaching the timeout period
+## Windows Service fails to start due to reaching the timeout period
 
-Sometimes when the server hosting a microservice that uses NServiceBus is started, the following exception is seen in windows event viewer.
-> A timeout was reached (30000 milliseconds) while waiting for the XYZ service to connect.
-> The XYZ service failed to start due to the following error:
-> The service did not respond to the start or control request in a timely fashion
+Sometimes, a Windows service that is set to start automatically will fail to start due to a dependency not yet being available before it times out. The following exception can be seen in the Windows Event Viewer:
 
-Yet, after a certain period of time, the service is able to start up again without any issues. This happens because during the restart, NServiceBus might still be waiting on some infrastructure to start up before it is able to initialize. This can be mitigated by:
-
-- Letting the dependencies of a service finish starting up and running before the service.
-- Setting the service to  "Automatic Delayed Start" so that it will only get the signal to start when all other “Automatic” services are running. This is because, when a windows service startup is set to "Automatic", it loads during boot whereas when it is set to "Automatic (delayed start)", it does not  start until after all other auto-start services have been launched. Once all the automatic start services are loaded, the system then queues the “delay start” services for 2 minutes (120 seconds) by default. This interval can be altered by creating a registry DWORD (32-bit) value named AutoStartDelay and setting the delay (base: decimal) in seconds, in the following registry key:
 ```txt
- HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control
+A timeout was reached (30000 milliseconds) while waiting for the XYZ service to connect.
+The XYZ service failed to start due to the following error:
+The service did not respond to the start or control request in a timely fashion
 ```
+
+Generally, the service will start at a later time without issue. 
+
+This problem can be mitigated by:
+
+- Configuring the service with the `depend= <dependencies>` parameter set which identifies which dependencies must start before this service starts. 

```

Comment:
```suggestion
- Enable service recoverability to ensure the service will auto restart in case start fails
- Configuring the service with the `depend= <dependencies>` parameter set which identifies which dependencies must start before this service starts. 

```